File: verification/cpp/section1_correction_b/main.cpp
#include <cmath>
#include <iostream>
#include <iomanip>
constexpr double PI = 3.14159265358979323846;
bool check(const std::string& n, double e, double a) {
    bool p = std::abs(e-a) < 1e-10;
    std::cout << "  [" << (p?"PASS":"FAIL") << "] " << n << "\n";
    return p;
}
int main() {
    std::cout << "=== Section 1: Correction b (C++) ===\n";
    double b = PI / (4*PI*PI + 2*PI*std::sqrt(3.0));
    double theta = std::asin(b);
    bool all = true;
    all &= check("b > 0", 1, b > 0 ? 1 : 0);
    all &= check("b < 1", 1, b < 1 ? 1 : 0);
    all &= check("sin(theta)=b", b, std::sin(theta));
    all &= check("cos^2+sin^2=1", 1.0, std::pow(std::cos(theta),2)+std::pow(std::sin(theta),2));
    std::cout << "JSON: {\"section\": 1, \"language\": \"cpp\", \"values\": {\"b\": " << b << "}, \"all_passed\": " << (all?"true":"false") << "}\n";
    return all ? 0 : 1;
}

File: verification/rust/section1_correction_b/src/main.rs
use std::f64::consts::PI;

fn compute_b() -> f64 { PI / (4.0 * PI * PI + 2.0 * PI * 3.0_f64.sqrt()) }

fn check(name: &str, expected: f64, actual: f64) -> bool {
    let pass = (expected - actual).abs() < 1e-10;
    println!("  [{}] {}: expected={:.15e}, actual={:.15e}",
        if pass {"PASS"} else {"FAIL"}, name, expected, actual);
    pass
}

fn main() {
    println!("=== Section 1: Correction b (Rust) ===");
    let b = compute_b();
    let theta = b.asin();
    let det_r = theta.cos() * theta.cos() + theta.sin() * theta.sin();

    let mut all_pass = true;
    all_pass &= check("b > 0", 1.0, if b > 0.0 {1.0} else {0.0});
    all_pass &= check("b < 1", 1.0, if b < 1.0 {1.0} else {0.0});
    all_pass &= check("sin(theta_b) = b", b, theta.sin());
    all_pass &= check("cos^2+sin^2 = 1", 1.0, det_r);

    println!("JSON: {{\"section\": 1, \"language\": \"rust\", \"values\": {{\"b\": {:.15e}}}, \"all_passed\": {}}}",
        b, all_pass);
    if !all_pass { std::process::exit(1); }
}
